app/Http/Controllers/LinkController.php

```diff
@@ -74,12 +81,16 @@ public function performRedirect(Request $request, $short_url, $secret_key=false)
         	if (!$secret_key) {
         		// if we do not receieve a secret key
         		// when we are expecting one, return a 403
-        		return abort(403);
+                return view('error', [
+                    'message' => 'Sorry, but this link needs a secret key.'
```

Comment:
Maybe "requires" is better language to use here. Perhaps returning a 404 is more appropriate, because users shouldn't necessarily be aware that a link requires a special ending ("secret key") to use.